src/codeu/chat/server/Controller.java

```diff
@@ -147,6 +164,67 @@ public Conversation newConversation(Uuid id, String title, Uuid owner, Time crea
     return conversation;
   }
 
+  // Search for a user in the database and create the User object in the return.
+  @Override
+  public User searchUserInDatabase(String username, String pswd){
+    DatabaseUser tempUser = databaseUsersMap.get(username);
+    //return (tempUser != null) ? (tempUser.pswd.equals(pswd)) ? model.userById().first(tempUser.id) : null : null;
+    if(tempUser != null){
+      if(tempUser.pswd.equals(pswd)){
+        final User userFromModel = model.userById().first(Uuids.fromString(tempUser.id));
+        if(userFromModel != null){
+          return userFromModel;
+        }else{
+          System.out.println("User in model but not in database");
+        }
+      }
+    }
+    return null;
+    // Iterate in every DatabaseUser
+    /*for(DatabaseUser databaseUser : databaseUsersList){
```

Comment:
Same as before, here's a bunch of commented code that we most likely don't need anymore and it should not be committed to the repository.